### PR TITLE
Update dependencies, ESLint configuration, and add pull request template

### PR DESCRIPTION
**Pull request by torosent**
# Summary

## What changed?
- Updated ESLint from v8 to v9.39.2 with new flat config format (`eslint.config.mjs`)
- Updated Prettier from v2 to v3.5.3
- Removed deprecated ESLint plugins and replaced with `typescript-eslint` v8.54.0
- Fixed unsafe stack trace access in `pb-helper.util.ts` (`e.stack?.toString() ?? ""`)
- Removed useless try-catch blocks in `client.ts`
- Removed sensitive data (activity output) from log messages in `activity-executor.ts`
- Fixed type imports in `worker-builder.ts` to use package exports instead of `any`
- Deleted duplicate/stale worker file `src/worker/task-hub-grpc-worker.ts`
- Updated example imports to use package exports instead of relative paths
- Fixed pre-existing tsconfig build issue by adding `baseUrl` to azuremanaged package
- Added PR template

## Why is this change needed?
- ESLint v8 is deprecated; v9 uses the new flat config format
- Code quality fixes identified during code review
- Standardize PR process with template

## Issues / work items
- Related #N/A

---

# Project checklist
- [x] Release notes are not required for the next release
- [x] Backport is not required
- [x] All required tests have been added/updated (unit tests, E2E tests)
- [x] Breaking change?
  - No breaking changes

---

# AI-assisted code disclosure (required)

## Was an AI tool used? (select one)
- [ ] No
- [x] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [ ] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): GitHub Copilot (Claude)
- AI-assisted areas/files: ESLint config migration, code fixes, PR template
- What you changed after AI output: Reviewed and validated all changes

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing

## Automated tests
- Result: Passed (138 tests: 84 in durabletask-js + 54 in durabletask-js-azuremanaged)

## Manual validation (only if runtime/behavior changed)
- N/A - No runtime behavior changes

---

# Notes for reviewers
- The duplicate file `src/worker/task-hub-grpc-worker.ts` was stale and used an older `promisify` pattern vs the current `callWithMetadata` approach
- ESLint now uses the flat config format (`.mjs` extension to avoid module warning)

### CODE REVIEW

File: packages/durabletask-js/src/worker/activity-executor.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export class ActivityExecutor {
     // Return the output
     const encodedOutput = activityOutput ? JSON.stringify(activityOutput) : undefined;
     const chars = encodedOutput ? encodedOutput.length : 0;
-    console.log(`Activity ${name} completed with output ${encodedOutput} (${chars} chars)`);
```
Comment (YunchuWang) [inline packages/durabletask-js/src/worker/activity-executor.ts]: i think with output log message is more clear

**Review by YunchuWang — APPROVED** [review]
